modules/artifact/src/main/java/org/mule/runtime/module/artifact/internal/descriptor/FeatureFlaggingUtils.java

```diff
@@ -4,7 +4,7 @@
  * license, a copy of which has been included with this distribution in the
  * LICENSE.txt file.
  */
-package org.mule.runtime.module.deployment.impl.internal.config;
+package org.mule.runtime.module.artifact.internal.descriptor;
```

Comment:
I had to move it from `deployment-impl` to `artifact` because we now would need to use it in the later. The `deployment-impl` module already has the `artifact` module as a dependency.
This one still sounds like a reasonable place to have this util class, because it deals mostly with `ArtifactDescriptor` which is an `artifact` class. For this same reason, putting this in `core` is not possible.